admin/pre_reg/views.py

```diff
@@ -20,169 +17,172 @@
-
+from admin.base.utils import PreregAdmin
+
+
+class DraftListView(PreregAdmin, ListView):
+    template_name = 'pre_reg/draft_list.html'
+    paginate_by = 10
+    paginate_orphans = 1
+    ordering = '-approval.initiation_date'
+    context_object_name = 'draft'
+
+    def get_queryset(self):
+        prereg_schema = MetaSchema.find_one(
+            Q('name', 'eq', 'Prereg Challenge') &
+            Q('schema_version', 'eq', 2)
+        )
+        query = (
+            Q('registration_schema', 'eq', prereg_schema) &
+            Q('approval', 'ne', None)
+        )
+        return DraftRegistration.find(query).sort(self.ordering)
+
+    def get_context_data(self, **kwargs):
+        query_set = kwargs.pop('object_list', self.object_list)
+        page_size = self.get_paginate_by(query_set)
+        paginator, page, query_set, is_paginated = self.paginate_queryset(
+            query_set, page_size)
+        return {
+            'drafts': [
+                serializers.serialize_draft_registration(d, json_safe=False)
+                for d in query_set
+            ],
+            'page': page,
+        }
+
+
+class DraftDetailView(PreregAdmin, DetailView):
+    template_name = 'pre_reg/draft_detail.html'
+    context_object_name = 'draft'
+
+    def get(self, request, *args, **kwargs):
+        try:
+            return super(DraftDetailView, self).get(request, *args, **kwargs)
+        except AttributeError:
+            return handle_attribute_error(request, self.context_object_name,
+                                          self.kwargs.get('draft_pk'))
 
-@csrf_exempt
-@user_passes_test(is_in_prereg_group)
-def approve_draft(request, draft_pk):
-    """Approves current draft
-    :param request: mostly for user
-    :param draft_pk: Unique id for current draft
-    :return: DraftRegistrationApproval obj
-    """
-    draft = get_draft_or_error(draft_pk)
-
-    user = request.user.osf_user
-    draft.approve(user)
-    update_admin_log(
-        request.user.id, draft._id, 'Draft Registration',
-        'approved', action_flag=ACCEPT_PREREG
-    )
-    return redirect(reverse('pre_reg:prereg') + "?page={0}".format(request.POST.get('page', 1)), permanent=True)
-
-
-@csrf_exempt
-@user_passes_test(is_in_prereg_group)
-def reject_draft(request, draft_pk):
-    """Rejects current draft
-    :param request: mostly for user
-    :param draft_pk: Unique id for current draft
-    :return: DraftRegistrationApproval obj
-    """
-    draft = get_draft_or_error(draft_pk)
-
-    user = request.user.osf_user
-    draft.reject(user)
-    update_admin_log(
-        request.user.id, draft._id, 'Draft Registration',
-        'rejected', action_flag=REJECT_PREREG
-    )
-    return redirect(reverse('pre_reg:prereg') + "?page={0}".format(request.POST.get('page', 1)), permanent=True)
+    def get_object(self, queryset=None):
+        return serializers.serialize_draft_registration(
+            DraftRegistration.load(self.kwargs.get('draft_pk'))
```

Comment:
Is serialize_draft_registration None safe? `DraftRegistration.load("something_coming_from_a_querystring_that's_not_real")` will return None.